src/routes/chat-with-tools/+page.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import { useChat } from '@ai-sdk/svelte';
+
+  const { input, handleSubmit, messages, addToolResult } = useChat({
+    api: '/api/use-chat-tools',
+    maxToolRoundtrips: 5,
+    // run client-side tools that are automatically executed:
+
+    async onToolCall({ toolCall }) {
+        if (toolCall.toolName === 'getLocation') {
+            const cities = ['New York', 'Los Angeles', 'Chicago', 'San Francisco'];
+            return cities[Math.floor(Math.random() * cities.length)];
+        }
+    }
+  });
+</script>
+
+<main>
+  <br />
+  <ul>
+      {#each $messages as message (message.id)}
+          <li>{message.role}: {message.content}</li>
+          {#if message.toolInvocations}
+              {#each message.toolInvocations as toolInvocation (toolInvocation.toolCallId)}
+                  {@const toolCallId = toolInvocation.toolCallId}
+
+                  {#if toolInvocation.toolName === 'askForConfirmation'}
+                      <div>
+                          {toolInvocation.args.message}
+                          <div>
+                              {#if 'result' in toolInvocation}
+                                  <b>{toolInvocation.result}</b>
+                              {:else}
+                                  <button on:click={() => addToolResult({ toolCallId, result: 'Yes' })}>Yes</button>
+                                  <button on:click={() => addToolResult({ toolCallId, result: 'No' })}>No</button>
+                              {/if}
+                          </div>
+                      </div>
+                  {/if}
+
+                  {#if 'result' in toolInvocation}
+                      <div>
+                          Tool call {`${toolInvocation.toolName}: `}
```

Comment:
tool calls are very interesting, will you use them in article? would be nice, otherwise remove

Reply:
this most probably will be removed in next iteration from the repo.